lib/MusicBrainz/Server/Controller/Role/Collection.pm

```diff
@@ -45,27 +56,39 @@ role
     method _stash_collections => sub {
         my ($self, $c) = @_;
 
-        my $collections;
+        my $own_collections;
+        my $collaborator_collections;
         my %containment;
-        my $entity_collections = $self->_all_collections($c);
+        my $entity_collections = $self->_all_visible_collections($c);
         my %entity_collections_map = map { $_->id => 1 } @$entity_collections;
 
+        my $number_of_visible_collections = @$entity_collections;
+        my $number_of_non_visible_collections = @{$self->_all_non_visible_collections($c)};
+
         if ($c->user_exists) {
             # Make a list of collections and whether this entity is contained in them
-            ($collections) = $c->model('Collection')->find_by({
+            ($own_collections) = $c->model('Collection')->find_by({
                 editor_id => $c->user->id,
                 entity_type => $entity_type,
                 show_private => $c->user->id,
             });
-            foreach my $collection (@$collections) {
+            foreach my $collection (@$own_collections) {
+                $containment{$collection->id} = 1 if $entity_collections_map{$collection->id};
+            }
+            ($collaborator_collections) = $c->model('Collection')->find_by({
+                collaborator_id => $c->user->id,
+                entity_type => $entity_type,
+            });
+            foreach my $collection (@$collaborator_collections) {
                 $containment{$collection->id} = 1 if $entity_collections_map{$collection->id};
             }
         }
 
         $c->stash
-          (collections => $collections,
+          (own_collections => $own_collections,
+           collaborator_collections => $collaborator_collections,
            containment => \%containment,
-           all_collections => $entity_collections,
+           number_of_collections => $number_of_visible_collections + $number_of_non_visible_collections,
```

Comment:
Something isn't right here, because if I'm on an entity which isn't in any collection, the collection section in the sidebar isn't displayed at all, so I can't add it to anything.